__tests__/unit/fact.test.js

```diff
@@ -0,0 +1,415 @@
+    expect(typeof invalidInputNum).toBe("number");
+  });
+
+  test("return random number when min and max are undefined", function () {
+    let noMinMax = getRandomApiNum({ type: "year" });
+    expect(typeof noMinMax).toBe("number");
+  });
+});
+
+describe("getRandomApiNum() with type 'math'", () => {
+  test("return same number as min and max", function () {
+    let sameAsMinMax = getRandomApiNum({ min: 2010, max: 2010, type: "math" });
+    expect(sameAsMinMax).toEqual(2010);
+  });
+
+  test("return random number greater than 2010", function () {
+    let greaterThanMin = getRandomApiNum({ min: 2010, type: "math" });
+    expect(greaterThanMin).toBeGreaterThan(2010);
+  });
+
+  test("return random number less than 2010", function () {
+    let lessThanMax = getRandomApiNum({ max: 2010, type: "math" });
+    expect(lessThanMax).toBeLessThan(2010);
+  });
+
+  test("return random number when min and max are NaN", function () {
+    let invalidInputNum = getRandomApiNum({ min: "a", max: "b", type: "year" });
+    expect(typeof invalidInputNum).toBe("number");
+  });
+
+  test("return random number when min and max are undefined", function () {
+    let noMinMax = getRandomApiNum({ type: "year" });
+    expect(typeof noMinMax).toBe("number");
+  });
+});
+
+describe("getRandomApiNum() with type 'year'", () => {
+  test("return same number as min and max", function () {
+    let sameAsMinMax = getRandomApiNum({ min: 2010, max: 2010, type: "year" });
+    expect(sameAsMinMax).toEqual(2010);
+  });
+
+  test("return random number greater than 2010", function () {
+    let greaterThanMin = getRandomApiNum({ min: 2010, type: "year" });
+    expect(greaterThanMin).toBeGreaterThan(2010);
+  });
+
+  test("return random number less than 2010", function () {
+    let lessThanMax = getRandomApiNum({ max: 2010, type: "year" });
+    expect(lessThanMax).toBeLessThan(2010);
+  });
+
+  test("return random number when min and max are NaN", function () {
+    let invalidInputNum = getRandomApiNum({ min: "a", max: "b", type: "year" });
+    expect(typeof invalidInputNum).toBe("number");
+  });
+
+  test("return random number when min and max are undefined", function () {
+    let noMinMax = getRandomApiNum({ type: "year" });
+    expect(typeof noMinMax).toBe("number");
+  });
+});
+
+describe("getSentence() for type 'year'", () => {
+  test("When wantFragment is true, returns partial text only", function () {
+    const sentence = getSentence({
+      wantFragment: { fragment: true },
+      number: 1000,
+      type: "year",
+      data: {
+        text: "Sweyn I establishes Danish control over part of Norway",
+      },
+    });
+    expect(sentence).toBe(
+      "Sweyn I establishes Danish control over part of Norway"
+    );
+  });
+
+  test("When wantFragment is undefined, returns full sentence", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
```

Comment:
you should be able to omit `wantFragment` entirely if you want its value to be `undefined`